x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/all_exception_items_table/index.tsx

```diff
@@ -0,0 +1,393 @@
+  listType: ExceptionListTypeEnum;
+  onRuleChange?: () => void;
+}
+
+const ExceptionsViewerComponent = ({
+  rule,
+  listType,
+  onRuleChange,
+}: ExceptionsViewerProps): JSX.Element => {
+  const { services } = useKibana();
+  const toasts = useToasts();
+  const [{ canUserCRUD, hasIndexWrite }] = useUserData();
+  const [isReadOnly, setReadOnly] = useState(true);
+  const [lastUpdated, setLastUpdated] = useState<null | string | number>(null);
+  const exceptionListsToQuery = useMemo(
+    () =>
+      rule != null && rule.exceptions_list != null
+        ? rule.exceptions_list.filter((list) => list.type === listType)
+        : [],
+    [listType, rule]
+  );
+
+  // Reducer state
+  const [{ exceptions, pagination, currenFlyout, exceptionToEdit, viewerState }, dispatch] =
+    useReducer(allExceptionItemsReducer(), {
+      ...initialState,
+    });
+
+  // Reducer actions
+  const setExceptions = useCallback(
+    ({
+      exceptions: newExceptions,
+      pagination: newPagination,
+    }: UseExceptionListItemsSuccess): void => {
+      setLastUpdated(Date.now());
+
+      dispatch({
+        type: 'setExceptions',
+        exceptions: newExceptions,
+        pagination: newPagination,
+      });
+    },
+    [dispatch]
+  );
+
+  const setViewerState = useCallback(
+    (state: ViewerState): void => {
+      dispatch({
+        type: 'setViewerState',
+        state,
+      });
+    },
+    [dispatch]
+  );
+
+  const setFlyoutType = useCallback(
+    (flyoutType: ViewerFlyoutName): void => {
+      dispatch({
+        type: 'updateFlyoutOpen',
+        flyoutType,
+      });
+    },
+    [dispatch]
+  );
+
+  const [_, allReferences] = useFindExceptionListReferences(exceptionListsToQuery);
+
+  const handleFetchItems = useCallback(
+    async (options?: GetExceptionItemProps) => {
+      const abortCtrl = new AbortController();
+
+      const newPagination =
+        options?.pagination != null
+          ? {
+              page: (options.pagination.page ?? 0) + 1,
+              perPage: options.pagination.perPage,
+            }
+          : {
+              page: pagination.pageIndex + 1,
+              perPage: pagination.pageSize,
+            };
+
+      if (exceptionListsToQuery.length === 0) {
+        return {
+          data: [],
+          pageIndex: pagination.pageIndex,
+          itemsPerPage: pagination.pageSize,
+          total: 0,
+        };
+      }
+
+      const {
+        page: pageIndex,
+        per_page: itemsPerPage,
+        total,
+        data,
+      } = await fetchExceptionListsItemsByListIds({
+        filter: undefined,
+        http: services.http,
+        listIds: exceptionListsToQuery.map((list) => list.list_id),
+        namespaceTypes: exceptionListsToQuery.map((list) => list.namespace_type),
+        search: options?.search,
+        pagination: newPagination,
+        signal: abortCtrl.signal,
+      });
+
+      // Please see `x-pack/plugins/lists/public/exceptions/transforms.ts` doc notes
```

Comment:
this file doesn't exist?

Reply:
Good catch. It does, looks like it was moved - I'll update in follow up. Trying to unblock my team.

Reply:
Thank you thank you for your feedback. I've opened a follow up PR addressing it [here](https://github.com/elastic/kibana/pull/140448).